src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -163,13 +172,62 @@ def _get_microsoft_graph_resource_id(cloud_name):
     return graph_endpoint_mapper.get(cloud_name, None)
 
 
+def _get_database_server_endpoint(db_prefix, sql_server_hostname, cloud_name):
+    if cloud_name == 'AzureCloud':
+        return db_prefix + '.database.azure.com'
```

Comment:
why do you need different logic for AzureCloud? why not just add '.database.azure.com' as its hostname in AzureCloud

Reply:
The data is based on the endpoints in https://management.azure.com/metadata/endpoints?api-version=2019-05-01 and `AzureCloud` uses "database.windows.net" for `sqlServerHostname`(may be outdated), need to change to `.database.azure.com`.

We'll need to update the data in the url to remove this patch.